hibernate-core/src/main/java/org/hibernate/action/internal/BulkOperationCleanupAction.java

```diff
@@ -91,7 +92,7 @@ public BulkOperationCleanupAction(SharedSessionContractImplementor session, Quer
 			}
 		}
 
-		this.affectedTableSpaces = spacesList.toArray( new String[ spacesList.size() ] );
+		this.affectedTableSpaces = spacesList.toArray( ArrayHelper.EMPTY_STRING_ARRAY );
```

Comment:
In modern JVM, `spacesList.toArray( new String[0] )` is faster than `spacesList.toArray( new String[ spacesList.size() ] )`. E.g. https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size

Similar changes are done elsewhere in this PR.